src/Npgsql/NpgsqlTypes/NpgsqlDate.cs

```diff
@@ -1,452 +1,235 @@
+        /// <value>The number of days since PostgreSQL epoch.</value>
+        /// <seealso cref="PostgreSqlEpoch"/>
+        public int Days => _days;
 
-        public static bool operator ==(NpgsqlDate x, NpgsqlDate y) => x.Equals(y);
-        public static bool operator !=(NpgsqlDate x, NpgsqlDate y) => !(x == y);
-        public static bool operator <(NpgsqlDate x, NpgsqlDate y)  => x.CompareTo(y) < 0;
-        public static bool operator >(NpgsqlDate x, NpgsqlDate y)  => x.CompareTo(y) > 0;
-        public static bool operator <=(NpgsqlDate x, NpgsqlDate y) => x.CompareTo(y) <= 0;
-        public static bool operator >=(NpgsqlDate x, NpgsqlDate y) => x.CompareTo(y) >= 0;
+        /// <summary>
+        /// Determines whether the specified value is finite.
+        /// </summary>
+        /// <param name="value">A date.</param>
+        /// <returns>
+        /// <see langword="true"/> if the specified value is finite;
+        /// otherwise, <see langword="false"/>.
+        /// </returns>
+        public static bool IsFinity(NpgsqlDate value) =>
+            value != NegativeInfinity &&
+            value != PositiveInfinity;
 
-        public static DateTime ToDateTime(NpgsqlDate date)
-        {
-            switch (date._type)
-            {
-            case InternalType.Infinity:
-            case InternalType.NegativeInfinity:
-                throw new InvalidCastException("Infinity values can't be cast to DateTime");
-            case InternalType.Finite:
-                try { return new DateTime(date._daysSinceEra * NpgsqlTimeSpan.TicksPerDay); }
-                catch { throw new InvalidCastException(); }
-            default:
-                throw new InvalidOperationException($"Internal Npgsql bug: unexpected value {date._type} of enum {nameof(NpgsqlDate)}.{nameof(InternalType)}. Please file a bug.");
-            }
-        }
+        /// <summary>
+        /// Determines whether the specified value is infinite.
+        /// </summary>
+        /// <param name="value">A date.</param>
+        /// <returns>
+        /// <see langword="true"/> if the specified value is infinite;
+        /// otherwise, <see langword="false"/>.
+        /// </returns>
+        public static bool IsInfinity(NpgsqlDate value) =>
+            value == NegativeInfinity ||
+            value == PositiveInfinity;
 
-        public static explicit operator DateTime(NpgsqlDate date) => ToDateTime(date);
+        /// <summary>
+        /// Converts the value of the current <see cref="NpgsqlDate"/> object to its equivalent
+        /// string representation using the specified format and the formatting conventions of the current culture.
+        /// </summary>
+        /// <returns></returns>
+        public override string ToString() =>
+            this == PositiveInfinity ? "infinity" :
+            this == NegativeInfinity ? "-infinity" :
+            $"{Year:D4}-{Month:D2}-{Day:D2}";
```

Comment:
Why not build a DateTime here and call `ToString("D")` on it, making this culture-aware and aligning with DateTime? [DateTime.ToString by default formats with `G`](https://docs.microsoft.com/en-us/dotnet/api/system.datetime.tostring?view=net-5.0#System_DateTime_ToString) (long timestamp format), so it would be consistent for us to format with `D` (long date format).

Reply:
You can try to do it and see what will happen if you invoke `ToString` on `MinValue` or `MaxValue`, or you can remember why am rewriting the types and that PostgreSQL has much wider ranges than BCL or the current provider specific (: